screenpipe-audio/src/stt.rs

```diff
@@ -367,11 +375,59 @@ pub async fn stt(
             debug!("device: {}, starting decoding process", audio_input.device);
             let segments = dc.run(&mel)?;
             debug!("device: {}, decoding complete", audio_input.device);
-            Ok(segments
-                .iter()
-                .map(|s| s.dr.text.clone())
-                .collect::<Vec<String>>()
-                .join("\n"))
+
+            let mut ranges: HashSet<String> = HashSet::new();
+            let token_regex = Regex::new(r"<\|\d{1,2}\.\d{1,2}\|>")?;
+            let mut transcript = String::from("");
+
+            let mut min_time: f32 = f32::MAX;
+            let mut max_time: f32 = f32::MIN;
+            let mut i = 0;
+            let segments_len = segments.len();
+            for segment in segments {
+                let mut text = segment.dr.text.clone();
+
+                // maybe not <|0.00|> but <|12.34|>
+                let mut start = text[..8].to_string();
```

Comment:
can you explain this part in plain english?

also i think we should really use overlap somehow

example:

bob talking for 45 seconds

screenpipe record 30 then 30 and whisper will lose quality because not having context from the previous audio chunk

wdyt?

Reply:
I'm pretty busy right now I'm commenting to confirm that I see this. I'll answer you and add some comments to the code tomorrow.